fairscale/experimental/optim/testrun.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Testing scaler
+"""
+
+import numpy as np
+import torch
+import torch.optim as optim
+import torch.nn as nn
+from torch.utils.data import Dataset, TensorDataset, DataLoader
+from torch.utils.data.dataset import random_split
+from dynamic_loss_scaler import DynamicLossScaler
+
+scaler = DynamicLossScaler()
+np.random.seed(42)
+x = np.random.rand(100, 1)
+y = 1 + 2 * x + .1 * np.random.randn(100, 1)
+
+# Shuffles the indices
+idx = np.arange(100)
+np.random.shuffle(idx)
+
+# Uses first 80 random indices for train
+train_idx = idx[:80]
+# Uses the remaining indices for validation
+val_idx = idx[80:]
+
+# Generates train and validation sets
+x_train, y_train = x[train_idx], y[train_idx]
+x_val, y_val = x[val_idx], y[val_idx]
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+torch.manual_seed(42)
+a = torch.randn(1, requires_grad=True, dtype=torch.float, device=device)
+b = torch.randn(1, requires_grad=True, dtype=torch.float, device=device)
+print(a, b)
+
+x_train_tensor = torch.from_numpy(x_train).float().to(device)
+y_train_tensor = torch.from_numpy(y_train).float().to(device)
+
+lr = 1e-1
+n_epochs = 1000
+
+loss_fn = nn.MSELoss(reduction='mean')
+
+optimizer = optim.SGD([a, b], lr=lr)
+
+for epoch in range(n_epochs):
+    yhat = a + b * x_train_tensor
```

Comment:
Shall we use a nn.Linear? It's the same thing but used in Models.

Reply:
yes. I will use nn.Linear.